packages/components/src/color-picker/inputs.js

```diff
@@ -0,0 +1,287 @@
+				l: Number( ( data.l && data.l ) || this.props.hsl.l ),
+				source: 'hsl',
+			} );
+		}
+	}
+
+	renderFields() {
+		const { disableAlpha = false } = this.props;
+		if ( this.state.view === 'hex' ) {
+			return (
+				<fieldset>
+					<legend className="screen-reader-text">
+						{ __( 'Color value in HEX' ) }
+					</legend>
+					<div className="components-color-picker__inputs-fields">
+						<Input
+							label="hex"
+							value={ this.props.hex }
+							onChange={ this.handleChange }
+						/>
+					</div>
+				</fieldset>
+			);
+		} else if ( this.state.view === 'rgb' ) {
+			return (
+				<fieldset>
+					<legend className="screen-reader-text">
+						{ __( 'Color value in RGB' ) }
+					</legend>
+					<div className="components-color-picker__inputs-fields">
+						<Input
+							label="r"
+							value={ this.props.rgb.r }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="255"
+						/>
+						<Input
+							label="g"
+							value={ this.props.rgb.g }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="255"
+						/>
+						<Input
+							label="b"
+							value={ this.props.rgb.b }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="255"
+						/>
+						{ disableAlpha ? null : (
+							<Input
+								label="a"
+								value={ this.props.rgb.a }
+								onChange={ this.handleChange }
+								type="number"
+								min="0"
+								max="1"
+								step="0.05"
+							/>
+						) }
+					</div>
+				</fieldset>
+			);
+		} else if ( this.state.view === 'hsl' ) {
+			return (
+				<fieldset>
+					<legend className="screen-reader-text">
+						{ __( 'Color value in HSL' ) }
+					</legend>
+					<div className="components-color-picker__inputs-fields">
+						<Input
+							label="h"
+							value={ Math.round( this.props.hsl.h ) }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="359"
+						/>
+						<Input
+							label="s"
+							value={ Math.round( this.props.hsl.s * 100 ) }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="100"
+						/>
+						<Input
+							label="l"
+							value={ Math.round( this.props.hsl.l * 100 ) }
+							onChange={ this.handleChange }
+							type="number"
+							min="0"
+							max="100"
+						/>
+						{ disableAlpha ? null : (
+							<Input
+								label="a"
+								value={ this.props.hsl.a }
+								onChange={ this.handleChange }
+								type="number"
+								min="0"
+								max="1"
+								step="0.05"
+							/>
+						) }
+					</div>
+				</fieldset>
+			);
+		}
+	}
+
+	render() {
+		return (
+			<div className="components-color-picker__inputs-wrapper">
+				{ this.renderFields() }
+				<div className="components-color-picker__inputs-toggle">
+					<IconButton
+						icon="arrow-down-alt2"
+						label={ __( 'Toggle input type' ) }
+						onClick={ this.toggleViews }
+					/>
```

Comment:
When toggling color format, there's no audible feedback about what just happened. Screen reader users would be forced to explore the UI to get an idea of what just changed. A simple `speak` message could help and give an useful feedback.

Reply:
Added in bbfe56a3bc2c24ab0ca1f5ee564d49013514ae10.